ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java

```diff
@@ -361,8 +361,15 @@ private void checkAndMaybeComplete() {
             new RouterException("DeleteOperation failed possibly because some replicas are unavailable",
                 RouterErrorCode.AmbryUnavailable));
       } else if (operationTracker.hasFailedOnNotFound()) {
-        operationException.set(
-            new RouterException("DeleteOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
+        if (operationTracker.getSuccessCount() > 0) {
```

Comment:
this is all good, but can you add some test cases just like the ttlupdate?

Reply:
Done in commit [796d906](https://github.com/linkedin/ambry/pull/2208/commits/796d906360d7bd66620014173d65915cdff39962)